[Dart/Flutter] Version 6 changes & additions (#3901)
getsentry/sentry-docs: src/platforms/dart/usage/advanced-usage.mdx

}
```

> ## Reporting bad HTTP requests as errors

Reviewer: ```suggestion
## Reporting Bad HTTP Requests as Errors
```

> The `SentryHttpClient` can also catch exceptions that may occur during requests
> such as [`SocketException`](https://api.dart.dev/stable/2.13.4/dart-io/SocketException-class.html)s.

Reviewer: ```suggestion
The `SentryHttpClient` can also catch exceptions that may occur during requests,
such as [`SocketExceptions`](https://api.dart.dev/stable/2.13.4/dart-io/SocketException-class.html).
```

Reviewer 2: we can't rename that, it's a class name

Reviewer: ah okay, i assumed it was a typo because it looked very odd. how about this instead:

```suggestion
The `SentryHttpClient` can also catch exceptions that may occur during requests — for example [`SocketException`](https://api.dart.dev/stable/2.13.4/dart-io/SocketException-class.html).
```

Reviewer 2: looks good, thanks

This is currently an opt-in feature. The following example shows how to enable it.

```dart
import 'package:sentry/sentry.dart';

var client = SentryHttpClient(captureFailedRequests: true);
try {
var uriResponse = await client.post('https://example.com/whatsit/create',
     body: {'name': 'doodle', 'color': 'blue'});
 print(await client.get(uriResponse.bodyFields['uri']));
} finally {
 client.close();
}
```

> Furthermore you can track HTTP requests which are considered bad by you. The
> following example shows how to do it. It captures exceptions for each request
> with a status code withing the range from 400 to 404 and also for 500.

Reviewer: ```suggestion
Furthermore you can track HTTP requests that you consider bad. In the
following example, exceptions are captured for each request
with a status code within the range of 400 to 404, and also for 500.
```

```dart
import 'package:sentry/sentry.dart';

var client = SentryHttpClient(
  failedRequestStatusCodes: [
    SentryStatusCode.range(400, 404),
    SentryStatusCode(500),
  ],
);

try {
var uriResponse = await client.post('https://example.com/whatsit/create',
     body: {'name': 'doodle', 'color': 'blue'});
 print(await client.get(uriResponse.bodyFields['uri']));
} finally {
 client.close();
}
```

> 

Reviewer: can you update the title below to:

Tips for Catching Errors
